int32 cluster_id
int32 size
event[] events